crate/operator/cratedb.py

```diff
@@ -164,3 +166,22 @@ async def is_cluster_healthy(
             except ProgrammingError as e:
                 logger.warning("Failed to run health check query", exc_info=e)
                 return False
+
+
+async def are_snapshots_in_progress(
+    conn_factory, logger: logging.Logger
+) -> Tuple[bool, Optional[str]]:
+    async with conn_factory() as conn:
+        async with conn.cursor() as cursor:
+            logger.info("Checking if there are running snapshots ...")
+            try:
+                await cursor.execute(
+                    f"SELECT stmt FROM {config.JOBS_TABLE} WHERE stmt "
```

Comment:
is there a reason, why you don't use the state of the sys.snapshots table?
https://crate.io/docs/crate/reference/en/4.4/admin/system-information.html#snapshots

Reply:
Yes, a very good reason - I didn't know/remember it! :) But this is much better, I will change.

Reply:
Update: `sys.snapshots` does not show _in progress_ snapshots - only finished ones. So I cannot use this.

Reply:
Since CrateDB 4.4.0 there would be a `snapshot_restore` table, see https://github.com/crate/crate/commit/473400e096eb08c2cf12d590075627b6fe904157

Reply:
Oh, thanks @mfussenegger! This is great. We might want to change it to use that in the future, once we have no clusters < 4.4.0. If no one minds, for now I'd like to keep the query from `sys.jobs`, as it works well enough.